libraries/standard/http/src/http_client.c

```diff
@@ -75,41 +76,40 @@ static HTTPStatus_t _addHeader( HTTPRequestHeaders_t * pRequestHeaders,
         /* Set this flag to backtrack in case of HTTP_INSUFFICIENT_MEMORY. */
         hasTrailingLine = 1u;
         pBufferCur -= HTTP_HEADER_LINE_SEPARATOR_LEN;
-        pRequestHeaders->headersLen -= HTTP_HEADER_LINE_SEPARATOR_LEN;
     }
 
     /* Check if there is enough space in buffer for additional header. */
-    toAddLen = fieldLen + HTTP_HEADER_FIELD_SEPARATOR_LEN + valueLen + \
-               HTTP_HEADER_LINE_SEPARATOR_LEN +                        \
+    toAddLen = fieldLen + HTTP_HEADER_FIELD_SEPARATOR_LEN + valueLen +
+               HTTP_HEADER_LINE_SEPARATOR_LEN +
                HTTP_HEADER_LINE_SEPARATOR_LEN;
 
     /* If we have enough room for the new header line, then write it to the header buffer. */
-    if( ( pRequestHeaders->headersLen + toAddLen ) <= pRequestHeaders->bufferLen )
+    if( ( initHeadersLen + toAddLen ) <= pRequestHeaders->bufferLen )
```

Comment:
Why cache the `pRequestHeaders->headersLen` locally instead of using it here directly?